lib/puppet/parser/scope.rb

```diff
@@ -798,6 +798,19 @@ def setvar(name, value, options = EMPTY_HASH)
     else
       table[name] = value
     end
+    # Assign the qualified name in the environment
+    # Note that Settings scope sets source to Boolean true.
+    #
+    # require 'byebug'; debugger if name == "fooo"
+    if source = self.source.is_a?(Puppet::Resource) && self.source.type == :hostclass
+      sourcename = source.name
```

Comment:
Unless I'm mistaking, the line above this one, assigns a boolean value to `self.source` (it's an `attr_accessor` so there's no difference between `source` and `self.source`). If that boolean is `true`, it's now asked for a name. That cannot work so I assume that there's no test that covers this piece of code.